modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTask.java

```diff
@@ -230,30 +223,28 @@ public MarketDataRequirements requirements() {
   @SuppressWarnings("unchecked")
   public CalculationResult execute(CalculationEnvironment scenarioData) {
     CalculationMarketData calculationData = new DefaultCalculationMarketData(scenarioData, marketDataMappings);
-    Result<?> result;
-
-    try {
-      if (function instanceof CalculationSingleFunction) {
-        Object value = ((CalculationSingleFunction<CalculationTarget, Object>) function).execute(target, calculationData);
-        result = value instanceof Result ? (Result<?>) value : Result.success(value);
-
-      } else if (function instanceof CalculationMultiFunction) {
-        ImmutableSet<Measure> measures = ImmutableSet.of(getMeasure());
-        Map<Measure, Result<ScenarioResult<?>>> map =
-            ((CalculationMultiFunction<CalculationTarget>) function).calculate(target, measures, calculationData);
-        if (!map.containsKey(getMeasure())) {
-          throw new IllegalStateException(Messages.format(
-              "Function '{}' did not return requested measure '{}'", function.getClass().getName(), getMeasure()));
-        }
-        result = map.get(getMeasure());
+    Result<?> result = Result.wrap(() -> calculate(calculationData));
+    Result<?> converted = convertToReportingCurrency(result, calculationData);
+    return CalculationResult.of(target, rowIndex, columnIndex, converted);
+  }
 
-      } else {
-        throw new IllegalStateException(Messages.format("Unknown function type '{}'", function.getClass().getName()));
-      }
-    } catch (RuntimeException ex) {
-      result = Result.failure(ex);
+  /**
+   * Calculates the result for the specified market data.
+   * 
+   * @param calculationData  the market data
+   * @return the result
+   */
+  private Result<?> calculate(CalculationMarketData calculationData) {
+    ImmutableSet<Measure> measures = ImmutableSet.of(getMeasure());
+    Map<Measure, Result<ScenarioResult<?>>> map = function.calculate(target, measures, calculationData);
```

Comment:
Is the intention to change `CalculationTask` and `CalculationResult` to handle multiple values in a later PR?